libcontainer/init_linux.go

```diff
@@ -300,11 +301,18 @@ func setupUser(config *initConfig) error {
 		return err
 	}
 
+	setgroups, err := ioutil.ReadFile("/proc/self/setgroups")
+	if err != nil && !os.IsNotExist(err) {
+		return err
+	}
+
 	// This isn't allowed in an unprivileged user namespace since Linux 3.19.
 	// There's nothing we can do about /etc/group entries, so we silently
 	// ignore setting groups here (since the user didn't explicitly ask us to
 	// set the group).
-	if !config.RootlessEUID {
+	allowSupGroups := !config.RootlessEUID && strings.TrimSpace(string(setgroups)) != "deny"
```

Comment:
I would prefer `== "allow"`.

Reply:
I've used the `!= deny` so that when `/proc/self/setgroups` doesn't exist (and setgroups is an empty string), `allowSupGroups` is still true.